forge/routes/api/user.js

```diff
@@ -97,4 +97,36 @@ module.exports = async function (app) {
         sharedUser.updateUser(app, request.session.User, request, reply, 'user')
         return reply // fix errors in tests "Promise may not be fulfilled with 'undefined' when statusCode is not 204" https://github.com/fastify/help/issues/627
     })
+
+    /**
+     * Delete user
+     * @name /api/v1/user/
+     * @static
+     * @memberof forge.routes.api.user
+     */
+    app.delete('/', {
+        preHandler: app.needsPermission('user:delete')
+    }, async (request, reply) => {
+        try {
+            const user = request.session.User
+            const deletedUser = {
+                id: user.id,
+                hashid: user.hashid,
+                username: user.username,
+                email: user.email
+            }
+            await user.destroy()
+            // Create an audit log entry for the deleted user
+            // NOTE: it is called as the system user (0) because the user
+            // is already deleted at this point
+            await app.auditLog.User.user.deleted(0, null, deletedUser)
+            reply.clearCookie('sid')
+            reply.send({ status: 'okay' })
+        } catch (err) {
+            const resp = { code: 'unexpected_error', error: err.toString() }
+            await app.auditLog.User.user.deleted(request.session.User, resp, request.session.User)
+            reply.code(400).send(resp)
+        }
+        return reply // avoid "Promise may not be fulfilled with 'undefined' when statusCode is not 204" https://github.com/fastify/help/issues/627
```

Comment:
Trying to figure out why this is needed, given we don't have to do this in any other route.

Reply:
Trying to cast my mind back to when I encountered the `"Promise may not be fulfilled"` problem.  I will be honest and state I did not try this route without the `return reply` (I added it knowing it wont break anything and _may_ solve a _possible_ issue.)

I will it and see what the tests comeback with.

Reply:
removed in [0503989](https://github.com/flowforge/flowforge/pull/1787/commits/05039891a2b7fb1b6a2597468cd518389284a85f)